src/animation/interpolants/SmoothInterpolant.js

```diff
@@ -0,0 +1,103 @@
+/**
+ *
+ * Cubic hermite spline interpolant.
+ *
+ *
+ * @author tschw
+ */
+
+THREE.SmoothInterpolant = function( times, values, stride, result ) {
+
+	THREE.Interpolant.call( this, times, values, stride, result );
+
+};
+
+Object.assign( THREE.SmoothInterpolant.prototype, THREE.Interpolant.prototype, {
+
+	constructor: THREE.SmoothInterpolant,
+
+	DefaultParameters: {
+
+		zeroVelocityAtStart: false,
+		zeroVelocityAtEnd: false
+
+	},
+
+	_intervalChanged: function( i1, t0, t1 ) {
+
+		var times = this.times,
+			iPrev = i1 - 2,
+			iNext = i1 + 1,
+
+			tPrev = times[ iPrev ],
+			tNext = times[ iNext ];
+
+		if ( tPrev === undefined ) {
+
+			iPrev = i1;
+
+			tPrev = this._getParameters().zeroVelocityAtStart ?
+					2 * t0 - t1 : // yields f'(t0) = 0, IOW accelerates
+					t1; // yields f''(t0) = 0, IOW turns into a straight line
+
+		}
+
+		if ( tNext === undefined ) {
+
+			iNext = i1 - 1;
+			tNext = this._getParameters().zeroVelocityAtEnd ?
+					2 * t1 - t0 : // yields f'(tN) = 0, IOW decelerates
+					t0; // yields f''(tN) = 0, IOW turns into a straight line
+
+		}
+
+		var halfDt = ( t1 - t0 ) * 0.5,
+			stride = this.stride;
+
+		this.weightPrev = halfDt / ( t0 - tPrev );
+		this.weightNext = halfDt / ( tNext - t1 );
+		this.offsetPrev = iPrev * stride;
+		this.offsetNext = iNext * stride;
+
+	},
+
+	_interpolate: function( i1, t0, t, t1 ) {
+
+		var times = this.times,
+			values = this.values,
+			stride = this.stride,
+			result = this.result,
+
+			o1 = i1 * stride,		o0 = o1 - stride,
+			oP = this.offsetPrev, 	oN = this.offsetNext,
+
+			wP = this.weightPrev, wN = this.weightNext,
+
+			p = ( t - t0 ) / ( t1 - t0 ),
+			pp = p * p,
+			ppp = pp * p;
+
+		// evaluate polynomials
```

Comment:
> I wonder if it is possible to pull out some of the interpolator code from Curve Utils and coodinate it with this? 

For curves, `t` can be chosen to be uniform, while for interpolation over time, `t` changes its scale all the time. There are certain curve types that are constructed by changing the scale of `t`, in general however, interpolation over time is more constrained than curve rendering.

Since I don't have tangents or weights as input, I multiplied the tangent computation directly into the matrix of polynomial coefficients, transforming its base to refer to the control points directly. The matrix depends on the scale of the intervals.

When the matrix is multiplied with the powers of the parameter, it spits out a vector of linear weights for the four control points.